data/tutorials/language/0lg_00_basic_datatypes.md

```diff
@@ -0,0 +1,808 @@
+```ocaml
+# String.to_bytes "hello";;
+- : bytes = Bytes.of_string "hello"
+```
+Operations on `bytes` values are provided by the [`Stdlib`](/api/Stdlib.html) and the [`Bytes`](/api/Bytes.html) modules. Only the function `Bytes.get` allows direct access to the characters contained in a byte sequence. There is no direct access operator on byte sequences.
+
+### Arrays & Lists
+
+#### Arrays
+
+Arrays are finite and fixed-sized sequences of values of the same type. Here are a couple of examples:
+```ocaml
+# [| 0; 1; 2; 3; 4; 5 |];;
+- : int array = [|0; 1; 2; 3; 4; 5|]
+
+# [| 'x'; 'y'; 'z' |];;
+- : char array = [|'x'; 'y'; 'z'|]
+
+# [| "foo"; "bar"; "baz" |];;
+- : string array = [|"foo"; "bar"; "baz"|]
+```
+
+Arrays may contain values of any type. Here arrays are `int array`, `char array`, and `string array`, but any type of data can be used in an array. Usually, `array` is said to be a polymorphic type. Strictly speaking, it is a type operator, and it accepts a type as a parameter (here `int`, `char`, and `string`) to form another type (those inferred here). This is the empty array.
+```ocaml
+# [||];;
+- : 'a array = [||]
+```
+Remember, `'a` ("alpha") is a type variable: a type parameter that will be replaced by another type.
+
+Like `string` and `bytes`, arrays support direct access, but the syntax is not the same.
+```ocaml
+# [| 'x'; 'y'; 'z' |].(2);;
+- : char = 'z'
+```
+
+Arrays are mutable, meaning they can't be extended or shortened, but each element may be updated.
+```ocaml
+# let letter = [| 'v'; 'x'; 'y'; 'z' |];;
+val letter : char array = [|'v'; 'x'; 'y'; 'z'|]
+
+# letter.(2) <- '3';;
+- : unit = ()
+
+# letter;;
+- : char array = [|'v'; 'x'; '3'; 'z'|]
+```
+
+Operations on arrays are provided by the [`Array`](/api/Array.html) module. There is a dedicated tutorial on Arrays.
+
+#### Lists
+
+As literals, lists are very much like arrays. Here are the same previous examples turned into lists.
+```ocaml
+# [ 0; 1; 2; 3; 4; 5 ];;
+- : int list = [0; 1; 2; 3; 4; 5]
+
+# [ 'x'; 'y'; 'z' ];;
+- : char list = ['x'; 'y'; 'z']
+
+# [ "foo"; "bar"; "baz" ];;
+- : string list = ["foo"; "bar"; "baz"]
+```
+
+Like arrays, lists are finite sequences of values of the same type. They are polymorphic, too. However, lists are extensible, immutable, and don't support direct access to all the values they contain. Lists play a central role in functional programming, so they have a [dedicated tutorial](/docs/lists).
+
+Operations on lists are provided by the [`List`](/api/List.html) module. The `List.append` function concatenates two lists. It can be used as an operator with the symbol `@`.
```

Comment:
Do you want to tell them how to pronounce `@`?

Reply:
Damm! I don't how to pronounce that!